modules/cell-type-ewings/resources/usr/bin/assign-celltypes.R

```diff
@@ -0,0 +1,151 @@
+  ),
+  make_option(
+    opt_str = c("--auc_table"),
+    type = "character",
+    default = NULL,
+    help = "Path to TSV file containing the AUC values to use as thresholds for assigning tumor cell states."
+  ),
+  make_option(
+    opt_str = c("--cell_state_markers_file"),
+    type = "character",
+    default = NULL,
+    help = "Path to TSV file with marker genes to use for labeling EWS-high proliferative cells.
+      Must contain a `cell_state` and `ensembl_gene_id` column.
+      Marker genes that have `cell_state == proliferative` will be used."
+  ),
+  make_option(
+    opt_str = c("--output_file"),
+    type = "character",
+    help = "Path to file where results will be saved"
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# make sure input files exist
+stopifnot(
+  "sce file does not exist" = file.exists(opt$sce_file),
+  "consensus cell types file does not exist" = file.exists(opt$consensus_celltype_file),
+  "AUCell results file does not exist" = file.exists(opt$aucell_results_file),
+  "auc_table file does not exist" = file.exists(opt$auc_table),
+  "cell_state_markers_file does not exist" = file.exists(opt$cell_state_markers_file)
+)
+
+# load SCE
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+})
+
+# read in files
+sce <- readr::read_rds(opt$sce_file)
+consensus_df <- readr::read_tsv(opt$consensus_celltype_file)
+aucell_results_df <- readr::read_tsv(opt$aucell_results_file)
+auc_threshold_df <- readr::read_tsv(opt$auc_table)
+cell_state_markers_df <- readr::read_tsv(opt$cell_state_markers_file)
+
+
+# assign normal cell types using consensus cell types --------------------------
+assignment_df <- consensus_df |>
+  dplyr::mutate(
+    custom_annotation = dplyr::if_else(
+      !is.na(consensus_ontology),
+      consensus_annotation,
+      "unknown"
+    )
+  )
+
+
+# assign tumor cells and tumor cell states -------------------------------------
+# pull out tumor cell states
+tumor_cell_states <- unique(auc_threshold_df$cell_type)
+
+# for each cell state get a list of cells that belong to that cell state using AUCell results
+# then update the existing custom annotation column to label those cells with that cell state
+tumor_cell_states |>
+  purrr::walk(\(state){
+    # get a list of all gene sets that should be used to classify that cell state
+    geneset_auc_list <- auc_threshold_df |>
+      dplyr::filter(cell_type == state) |>
+      dplyr::select(gene_set, auc_threshold) |>
+      tibble::deframe()
+
+    # get cells that meet all criteria, e.g., have auc > threshold for all gene sets
+    cells <- geneset_auc_list |>
+      purrr::imap(\(auc_threshold, name){
+        aucell_results_df |>
+          dplyr::filter(gene_set == name & auc > auc_threshold) |>
+          dplyr::pull(barcodes)
+      }) |>
```

Comment:
Pretty minor, but you can avoid the deframe and just use `purrr::pmap()`. I added a little rename to the `select` to make the later comparison unambiguous.

```suggestion
    geneset_auc_df <- auc_threshold_df |>
      dplyr::filter(cell_type == state) |>
      dplyr::select(gs_name = gene_set, auc_threshold)

    # get cells that meet all criteria, e.g., have auc > threshold for all gene sets
    cells <- geneset_auc_df |>
      purrr::pmap(\(gs_name, auc_threshold){
        aucell_results_df |>
          dplyr::filter(gene_set == gs_name & auc > auc_threshold) |>
          dplyr::pull(barcodes)
      }) |>
```